sklearn/tree/_tree.pyx

```diff
@@ -867,6 +865,166 @@ cdef class Tree:
 
         return out
 
+    cpdef object decision_paths(self, object X):
```

Comment:
Might want to mention that it returns a sparse matrix, and the reason (because it has to have as many columns as the maximal path)